spec/controllers/registrations_controller_spec.rb

```diff
@@ -7,19 +7,30 @@
   end
 
   describe 'POST #create' do
+    let(:list_id) { ENV['MAILCHIMP_LIST_ID'] }
+    let(:email) { 'kevin@bazfiz.com' }
     let(:user_attributes) {
       {
         username: 'kevin',
-        email: 'kevin@example.com',
+        email: email,
         password: 'foobar1',
         password_confirmation: 'foobar1',
       }
     }
 
+    before do
+      mailchimp_remove_member(email, list_id)
+    end
+
     it 'redirects to the dashboard' do
       post :create, params: { user: user_attributes }
       expect(response).to redirect_to(dashboard_path)
     end
+
+    it 'registers the new user on the mailchimp mailing list', :vcr do
+      post :create, params: { user: user_attributes }
```

Comment:
can we stub out the call to your service object instead please?

Something like:

```
it 'registers the new user on the mailchimp mailing list' do
  expect(MailchimpSubscription).to receive(:create).
    with(email: email, username: 'kevin', signup_date: DateTime.now)

  post :create, params: { user: user_attributes }
end
```

The signup date might be trickier to stub.

Reply:
I'll try this out. Seems like it should work fine. If the unit test for `MailchimpSubscription` tests API interaction then there's no reason to hit the API here.